import/qml/AudioPlayer.qml

```diff
@@ -207,64 +276,137 @@ Item {
+                Layout.alignment: Qt.AlignHCenter
+                color: "transparent"
+
+                Row {
+                    id: repRows
+                    height: parent.height
+                    anchors.horizontalCenter: parent.horizontalCenter
+                    spacing: 4
+                    visible: spectrumVisible
+                    enabled: spectrumVisible
+                    z: -5
+
+                    Repeater {
+                        id: rep
+                        model: root.soundModelLength - 1
+
+                        delegate: Rectangle {
+                            width: (spectrumAreaCentered.width - (repRows.spacing * root.soundModelLength)) / root.soundModelLength
+                            radius: 3
+                            opacity: root.currentState === MediaPlayer.PlayingState ? 1 : 0
+                            height: 15 + root.spectrum[modelData] * root.spectrumHeight
+                            anchors.bottom: parent.bottom
+
+                            gradient: Gradient {
+                                GradientStop {position: 0.05; color: height > root.spectrumHeight / 1.25 ? spectrumColorPeak : spectrumColorNormal}
+                                GradientStop {position: 0.25; color: spectrumColorMid}
+                                GradientStop {position: 0.50; color: spectrumColorNormal}
+                                GradientStop {position: 0.85; color: spectrumColorMid}
+                            }
+
+                            Behavior on height {
+                                NumberAnimation {
+                                    duration: 150
+                                    easing.type: Easing.Linear
+                                }
+                            }
+                            Behavior on opacity {
+                                NumberAnimation{
+                                    duration: 1500 + root.spectrum[modelData] * parent.height
+                                    easing.type: Easing.Linear
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+
             RowLayout {
                 spacing: Kirigami.Units.smallSpacing
                 Layout.fillWidth: true
                 visible: root.progressBar ? 1 : 0
                 enabled: root.progressBar ? 1 : 0
 
-                Controls.Slider {
+                T.Slider {
                     id: seekableslider
-                    to: player.duration
+                    //to: root.playerDuration
                     Layout.fillWidth: true
+                    Layout.preferredHeight: Mycroft.Units.gridUnit
                     property bool sync: false
+                    value: root.playerPosition
+
+                    handle: Item {
+                        x: seekableslider.visualPosition * (parent.width - (Kirigami.Units.largeSpacing + Kirigami.Units.smallSpacing))
```

Comment:
I've been playing with this a little as I've been seeing the positionMarker move slightly beyond the end of the seek bar at the end of tracks. 

This seems to work for me: 
```
x: seekableslider.visualPosition / (parent.width + {implicitWidth_of_hand}) * parent.width ** 2
```
maybe easier to reason as:
```
(visualPosition * parent.width) / (parent.width + hand.width) * parent.width
```

Does that make sense?